cmd/dptp-controller-manager/main.go

```diff
@@ -317,27 +326,29 @@ func main() {
 	}
 	pjutil.ServePProf(flagutil.DefaultPProfPort)
 
-	// Needed by the ImageStreamTagReconciler. This is a setting on the SharedInformer
-	// so its applied for all watches for all controllers in this manager. If needed,
-	// we can move this to a custom sigs.k8s.io/controller-runtime/pkg/source.Source
-	// so its only applied for the ImageStreamTagReconciler.
-	// TODO alvaroalmean: This is crap. Add a proper time-based trigger on controller-level,
-	// not a global one for everything because one controller happens to need it.
-	resyncInterval := 24 * time.Hour
-	registryMgr, err := controllerruntime.NewManager(kubeconfigs[apiCIContextName], controllerruntime.Options{
-		LeaderElection: false,
-		// The normal manager already serves these metrics and we must disable it here to not
-		// get an error when attempting to create the second listener on the same address.
-		MetricsBindAddress: "0",
-		SyncPeriod:         &resyncInterval,
-		DryRunClient:       opts.dryRun,
-		Logger:             ctrlruntimelog.NullLogger{},
-	})
-	if err != nil {
-		logrus.WithError(err).Fatal("Failed to construct manager for registry")
-	}
-	if err := mgr.Add(registryMgr); err != nil {
-		logrus.WithError(err).Fatal("Failed to add registry manager to main manager.")
+	registryMgr := mgr
+	if !registryMigrated {
```

Comment:
Please don't use opaque and current-state-of-the-world based variables like `registryMigrated`. Iterate over all configs, construct a manager for each, if name == registryClusterName then you add the `SyncPeriod` to the options and set the `registryManager` variable

Reply:
Updated:
Had to loop twice:
* create the main manager for app.ci in the first loop
* add the others to the main manager in the 2nd loop